gcp-observability/src/test/java/io/grpc/gcp/observability/logging/GcpLogSinkTest.java

```diff
@@ -98,32 +100,31 @@ public class GcpLogSinkTest {
       .putFields("rpc_id", Value.newBuilder().setStringValue(rpcId).build())
       .build();
   private Logging mockLogging;
+  private GcpLogSink spySink;
 
   @Before
   public void setUp() {
     mockLogging = mock(Logging.class);
+    spySink = spy(new GcpLogSink(destProjectName, locationTags,
+        customTags, flushLimit));
```

Comment:
This code brings up a couple of points:

- it looks a little weird that `GcpLogSink` is the class-under-test and that class itself is being mocked or spied. Usually all the other external classes are mocked.

- overloading `createLoggingClient()` in a subclass would have been an option then I was wondering why `GcpLogSink` is not final. Should it not be (even if it makes testing difficult)?

Reply:
Updated to use test only constructor